ghga_connector/core/exceptions.py

```diff
@@ -37,52 +44,84 @@ def __init__(self, url: str):
         super().__init__(message)
 
 
-class NoS3AccessMethod(RuntimeError):
+class NoS3AccessMethod(RuntimeError, GHGAConnectorException):
     """Thrown, when a request returns the desired response code, but no S3 Access Method"""
 
     def __init__(self, url: str):
         message = f"The request to {url} did not return an S3 Access Method."
         super().__init__(message)
 
 
+class FileNotRegisteredError(RuntimeError, GHGAConnectorException):
+    """Thrown, when a request for a file returns a 404 error."""
+
+    def __init__(self, file_id: str):
+        message = (
+            f"The request for the file {file_id} failed, "
+            "because this file id does not exist."
+        )
+        super().__init__(message)
+
+
+class UploadNotRegisteredError(RuntimeError, GHGAConnectorException):
+    """Thrown, when a request for a multipart upload returns a 404 error."""
+
+    def __init__(self, upload_id: str):
+        message = (
+            f"The request for the upload with the id '{upload_id}' failed, "
+            "because this upload does not exist."
+        )
+        super().__init__(message)
+
+
 class BadResponseCodeError(RuntimeError):
-    """Thrown, when a request returns a non-desired response code (e.g. 400)"""
+    """Thrown, when a request returns a not expected response code (e.g. 500)"""
 
     def __init__(self, url: str, response_code: int):
         message = f"The request to {url} failed with response code {response_code}"
         super().__init__(message)
 
 
-class CantCancelUploadError(RuntimeError):
-    """Thrown, when a request returns a non-desired response code (e.g. 400)"""
+class UploadNotExistsError(RuntimeError, GHGAConnectorException):
```

Comment:
```suggestion
class UploadDoesNotExistError(RuntimeError, GHGAConnectorException):
```
Name sounds weird - `UploadDoesNotExistError` is not that much longer